lib/l10n_utils/fluent.py

```diff
@@ -0,0 +1,80 @@
+from functools import wraps
+from hashlib import sha256
+
+from django.conf import settings
+from django.core.cache import caches
+from django.utils.encoding import force_bytes
+
+from fluent.runtime import FluentLocalization, FluentResourceLoader
+
+
+cache = caches['l10n']
+
+
+class FluentL10n(FluentLocalization):
+    def has_message(self, message_id):
```

Comment:
@Pike I moved that logic here. I was going to suggest it as an addition to your fallback PR but I'm not sure the whole `if bundle.locales[0] != 'en'` is appropriate for a wider audence, so it seemed like a bedrock specific need. If you think you can pull it in then great.